File: tags.py
from google.cloud import vision_v1
from google.cloud.vision_v1 import types
import os
import io

#find current directory
cwd = os.getcwd()
api_key = cwd + "\quikShop.json"
image = cwd + "\image1.jpg"

#set api_key
os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = api_key

global tags
tags = ''

def detect_logos(path):
    """Detects logos in the file."""
    from google.cloud import vision_v1

    client = vision_v1.ImageAnnotatorClient()

    with io.open(path, 'rb') as image_file:
        content = image_file.read()

    image = vision_v1.types.Image(content=content)

    response = client.logo_detection(image=image)
    logos = response.logo_annotations
    print('Logos:')

    global tags

    for logo in logos:
        tags = tags + logo.description + ', ' 

def detect_labels(path):
    """Detects labels in the file."""
    from google.cloud import vision_v1
    
    client = vision_v1.ImageAnnotatorClient()

    with io.open(path, 'rb') as image_file:
        content = image_file.read()

    image = vision_v1.types.Image(content=content)

    response = client.label_detection(image=image)
    labels = response.label_annotations
    print('Labels:')

    global tags

    for label in labels:
        tags = tags + label.description + ', ' 

def detect_text(path):
    """Detects text in the file."""
    from google.cloud import vision_v1
    
    
    client = vision_v1.ImageAnnotatorClient()

    with io.open(path, 'rb') as image_file:
        content = image_file.read()

    image = vision_v1.types.Image(content=content)

    response = client.text_detection(image=image)
    texts = response.text_annotations
    print('Labels:')

    global tags

    for text in texts:
        tags = tags + text.description + ', ' 

detect_labels(image)
detect_logos(image)
detect_text(image)



print(tags)

file = open("tags.txt", "w+")
file. write(tags)
file.close()

File: tags.js
// Imports the Google Cloud client library.
const Storage = require('@google-cloud/storage');

// Imports the Google Cloud client library
const vision = require('@google-cloud/vision');

//define key variables
var cwd = process.cwd();

var key = cwd + '/quikShop.json';
var image = cwd + '/image1.jpeg';
var done = 0

var tags = []

GOOGLE_APPLICATION_CREDENTIAL = key;

function convert(array, target){
  var target = array.toString()
  var target = target.replace(/\r/g, "").replace(/\n/g, "");
  console.log(target)

}

const client = new vision.ImageAnnotatorClient();

//execute image calculation
client
  .labelDetection(image)
  .then(results => {
    const labels = results[0].labelAnnotations;
    console.log('Labels:');
    labels.forEach(label => {tags.push(label.description)});
    
    console.log(tags)

    if(done != 2){
      done = done + 1
    }
    else{
      convert(tags, tags)
    }
  })
  .catch(err => {
    console.error('ERROR:', err);
  });


client
  .logoDetection(image)
  .then(results => {
    const logos = results[0].logoAnnotations;
    console.log('Logos:');
    logos.forEach(logo => {tags.push(logo.description)});
    
    console.log(tags)

    if(done != 2){
      done = done + 1
    }
    else{
      convert(tags, tags)
    }
  })
  .catch(err => {
    console.error('ERROR:', err);
  });

client
  .textDetection(image)
  .then(results => {
    const detections = results[0].textAnnotations;
    console.log('Text:');
    detections.forEach(text => {tags.push(text.description)});
    
    console.log(tags)

    if(done != 2){
      done = done + 1
    }
    else{
      convert(tags, tags)
    }
  })
  .catch(err => {
    console.error('ERROR:', err);
  });